src/geom/transformFactory.cc

```diff
@@ -0,0 +1,130 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <sstream>
+#include <cmath>
+
+#include "astshim.h"
+
+#include "Eigen/Core"
+
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/transformFactory.h"
+#include "lsst/pex/exceptions.h"
+
+#include "ndarray.h"
+#include "ndarray/eigen.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+namespace {
+/**
+ * Print a vector to a stream.
+ *
+ * The exact details of the representation are unspecified and subject to
+ * change, but the following may be regarded as typical:
+ *
+ *     [1.0, -3.560, 42.0]
+ *
+ * @tparam T the element type. Must support stream output.
+ */
+template <typename T>
+std::ostream &operator<<(std::ostream &os, std::vector<T> const &v) {
+    os << '[';
+    bool first = true;
+    for (T element : v) {
+        if (first) {
+            first = false;
+        } else {
+            os << ", ";
+        }
+        os << element;
+    }
+    os << ']';
+    return os;
+}
+
+/**
+ * Convert a Matrix to the equivalent ndarray.
+ *
+ * @param matrix The matrix to convert.
+ * @returns an ndarray containing a copy of the data in `matrix`
+ */
+template <class T, int Rows, int Cols>
+ndarray::Array<T, 2, 2> toNdArray(Eigen::Matrix<T, Rows, Cols> const &matrix) {
+    ndarray::Array<T, 2, 2> array = ndarray::allocate(ndarray::makeVector(matrix.rows(), matrix.cols()));
```

Comment:
No, actually. `Rows` and `Cols` may have the dummy value -1 to mean "determined at run-time", and in fact they do have this value for the matrix returned by `getJacobian`. (Also, I'm pretty sure that in fixed-size matrices the implementations for `rows()` and `cols()` are optimized for inlining.)